examples/finite_prime_field.rs

```diff
@@ -0,0 +1,256 @@
+use noether::{
+    AssociativeAddition, AssociativeMultiplication, CommutativeAddition, CommutativeMultiplication,
+    DistributiveAddition, DistributiveMultiplication, FiniteField,
```

Comment:
Since distributivity always involves both operations, not clear how a distinction would be useful. Also, left and right distributivity are technically distinct but I'm not familiar with structures where only one would hold.

Reply:
That's true, could get rid of the separate marker traits likely.